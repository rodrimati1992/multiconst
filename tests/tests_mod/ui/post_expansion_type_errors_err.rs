type Pair = (u32, u32);


// ensure that the type mismatch errors point to the expression.
multiconst::multiconst!{
    const A000: u32 = ();
}


// ensure that mismatched tuple pattern and tuple type alias type error at macro expansion time
multiconst::multiconst!{
    const (): Pair = (3, 3);
}


// ensure that too long array patterns error 
multiconst::multiconst!{
    const [A1, B1, C1]: [(); 2] = [(),()];
}

// ensure that too long array patterns error 
multiconst::multiconst!{
    const [A11, B11, C11, ..]: [(); 2] = [(),()];
}

// ensure that too short array patterns error 
multiconst::multiconst!{
    const [A2, B2, C2]: [(); 4] = [(),(),(),()];
}



multiconst::multiconst!{
    const [X, Z, Z]: [(); 3] = [(), (), ()];
}

multiconst::multiconst!{
    const Z: () = ();
}



// ensure that type annotations cause a type mismatch when they should
multiconst::multiconst!{
    const _: () = 100;
}

// ensure that type annotations cause a type mismatch when they should
multiconst::multiconst!{
    const std::ops::Range{start: _: u32, ..}: std::ops::Range<u8> = 3u8..5;
}

// ensure that type annotations cause a type mismatch when they should
multiconst::multiconst!{
    const std::ops::Range{start: [_, _]: [u32; 2], ..}: std::ops::Range<[u8; 2]> = 
        [0u8; 2]..[0u8; 2];
}


fn main(){}