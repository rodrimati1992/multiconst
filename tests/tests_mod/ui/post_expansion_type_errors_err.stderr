error[E0428]: the name `Z` is defined multiple times
  --> tests/tests_mod/ui/post_expansion_type_errors_err.rs:34:18
   |
34 |     const [X, Z, Z]: [(); 3] = [(), (), ()];
   |               -  ^ `Z` redefined here
   |               |
   |               previous definition of the value `Z` here
   |
   = note: `Z` must be defined only once in the value namespace of this module
   = note: this error originates in the macro `$crate::__::__priv_multiconst_proc_macro` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0428]: the name `Z` is defined multiple times
  --> tests/tests_mod/ui/post_expansion_type_errors_err.rs:38:11
   |
34 |     const [X, Z, Z]: [(); 3] = [(), (), ()];
   |               - previous definition of the value `Z` here
...
38 |     const Z: () = ();
   |           ^ `Z` redefined here
   |
   = note: `Z` must be defined only once in the value namespace of this module
   = note: this error originates in the macro `$crate::__::__priv_multiconst_proc_macro` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
 --> tests/tests_mod/ui/post_expansion_type_errors_err.rs:6:23
  |
6 |     const A000: u32 = ();
  |                 ---   ^^ expected `u32`, found `()`
  |                 |
  |                 expected due to this

error[E0308]: mismatched types
  --> tests/tests_mod/ui/post_expansion_type_errors_err.rs:12:11
   |
12 |     const (): Pair = (3, 3);
   |           ^^  ---- expected due to this
   |           |
   |           expected tuple, found `()`
   |
   = note:  expected tuple `(u32, u32)`
           found unit type `()`

error[E0527]: pattern requires 3 elements but array has 2
  --> tests/tests_mod/ui/post_expansion_type_errors_err.rs:18:11
   |
18 |     const [A1, B1, C1]: [(); 2] = [(),()];
   |           ^^^^^^^^^^^^ expected 2 elements

error[E0528]: pattern requires at least 3 elements but array has 2
  --> tests/tests_mod/ui/post_expansion_type_errors_err.rs:23:11
   |
23 |     const [A11, B11, C11, ..]: [(); 2] = [(),()];
   |           ^^^^^^^^^^^^^^^^^^^ pattern cannot match array of 2 elements

error[E0527]: pattern requires 3 elements but array has 4
  --> tests/tests_mod/ui/post_expansion_type_errors_err.rs:28:11
   |
28 |     const [A2, B2, C2]: [(); 4] = [(),(),(),()];
   |           ^^^^^^^^^^^^ expected 4 elements
